### PR TITLE
Tests: some cleaning

### PR DESCRIPTION
**Pull request by vbgl**
I wished all test cases could be automatically verified for safety. It’s not the case yet, but this PR is a step towards this goal.

CI: https://gitlab.com/jasmin-lang/jasmin/-/pipelines/758179917 (failed…)

### CODE REVIEW

**Comment by vbgl** [comment]
CI: https://gitlab.com/jasmin-lang/jasmin/-/pipelines/758207824

File: compiler/tests/success/cmove8.jazz [hunk 1]
```diff
@@ -1,51 +1,47 @@
 export
 fn f(reg u8 b, reg u64 x y) -> reg u64 {
   reg u64 r;
-  reg u8 z;
   reg bool c;
 
   r = x;
   c = b == 1;
   r = y if c;
-  r = x if c;
```
Comment (eponier) [inline compiler/tests/success/cmove8.jazz:10]: Why did you remove all these lines?

File: compiler/tests/success/divmod.jazz [hunk 2]
```diff
@@ -6,6 +6,9 @@ reg u32 e, f;
 reg u16 g, h;
 
 r = 0;
+a = 1;
+x = a if x <=u 0;
+x = a if x >=u 32768;
```
Comment (eponier) [inline compiler/tests/success/divmod.jazz:11]: Why did you add this?

File: compiler/tests/success/pointers/test_spill_loop.jazz [hunk 3]
```diff
@@ -8,7 +8,7 @@ fn loop(reg u64 x) -> reg u64 {
    p = s;
    sp = p;
    y = 0;
-   while (y < x) {
+   while (y < 42) {
```
Comment (eponier) [inline compiler/tests/success/pointers/test_spill_loop.jazz:11]: Why this change?

**Review by eponier — COMMENTED** [review]
Some of these changes are obviously good: initialization of a previously non initialized var, aligned read or write to external memory. But I don't understand the other ones. What is the idea of those other changes?

File: compiler/tests/success/cmove8.jazz (hunk 1 above)
Comment (vbgl) [reply compiler/tests/success/cmove8.jazz:10]: Two reasons: the official one is that these lines bring nothing, they are redundant and do not add anything to the test case; the officious one is that there are some exponential behaviors in the safety checker, and the shorter test case can be checked in some reasonable amount of time.

File: compiler/tests/success/pointers/test_spill_loop.jazz (hunk 3 above)
Comment (vbgl) [reply compiler/tests/success/pointers/test_spill_loop.jazz:11]: To ensure termination.

**Review by bgregoir — APPROVED** [review]
